File: Transportation-model/src/main/java/com/transportation/transportation/model/entities/BusLine.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.transportation.transportation.model.entities;

/**
 *
 * @author youssef
 */
public class BusLine extends TransportationLine {

    public BusLine() {
    }

    public BusLine(TransportationLine transportationLine) {
        super(transportationLine);
    }

    @Override
    public TransportationLine clone() {
        return new BusLine(this);
    }

}

File: DartMobileClient/lib/model/bus_line.dart
part of model;

class BusLine extends TransportationLine {

  BusLine();

  BusLine.fromMap(Map busLineMap) : super.fromMap(busLineMap);

  Map toJson() {
    Map json = super.toJson();
    json["@type"] = "BusLine";
    return json;
  }
  
}
